### ISSUE TITLE
Wording of "hint locality" warning / error

### ISSUE THREAD

**Issue by MSoegtropIMC** [coq/coq#17309]
#### Description of the problem

Since 8.17+rc1 the "hint locality" warning is a fatal error by default. The wording of the message is not really appropriate for an error and should be adjusted:

```
Error: The default value for hint locality is currently "local" in a section
and "global" otherwise, but is scheduled to change in a future release. For
the time being, adding hints outside of sections without specifying an
explicit locality attribute is therefore deprecated. It is recommended to use
"export" whenever possible. Use the attributes #[local], #[global] and
#[export] depending on your choice. For example: "#[export] Hint Unfold foo :
bar." [deprecated-hint-without-locality,deprecated]
```

#### Coq Version

```
The Coq Proof Assistant, version 8.17+rc1
compiled with OCaml 4.13.1
```

**Comment by SkySkimmer** [comment]
It seems OK to me, do you have a specific change suggestion?

**Comment by MSoegtropIMC** [comment]
How about:

The default value for hint locality is currently "local" in a section and "global" otherwise, but is scheduled to change in a future release. Adding hints outside of sections without specifying an explicit locality attribute is therefore deprecated and results in a fatal error, which for the time being can be turned into a warning with command line option XYZ. It is recommended to use "export" whenever possible. Use the attributes #[local], #[global] and
#[export] depending on your choice. For example: "#[export] Hint Unfold foo :
bar." [deprecated-hint-without-locality,deprecated]

**Comment by ppedrot** [comment]
This is inaccurate, because the ability to not put an attribute will be (actually has been) reintroduced. So it's not deprecated for ever, but just *for the time being*.

**Comment by MSoegtropIMC** [comment]
I see. What is the rationale then to make this a fatal error by default in 8.17+rc1?

As far as I can see this is the #1 build breaker in Coq Platform 2022.03.

**Comment by ppedrot** [comment]
It's to ensure that we have a released version where the change is intentionally breaking. You *have* to port your code.

**Comment by MSoegtropIMC** [comment]
OK, but then again I don't think that the message is appropriate.

**Comment by Zimmi48** [comment]
The message could say something like "This warning is set as fatal by default to ensure that all hint localities are made explicit before their default value changes in Coq 8.18."

@MSoegtropIMC @ppedrot What do you think?

**Comment by ppedrot** [comment]
This looks fine to me.

**Comment by Zimmi48** [comment]
I'm going to prepare a PR.

**Comment by MSoegtropIMC** [comment]
@Zimmi48 : yes, mentioning that it is a warning which has been made fatal by default is a good solution.

**Comment by Zimmi48** [comment]
Fixed by merging #17343 in v8.17.
